File: js/intro.js
"use strict";

// Port of intro.java.

class IntroLevel extends KLevel {
  constructor(game) {
    super(game);
    this.plot = "Willkommen!";
    this.inst = "Click the NEXT button to go to the next level.";

    this.yshift = [-6, -6, -5, -4, -3, -2, -1, 0, 0, -1, -2, -3, -4, -5, -6];
    this.cols = [130, 140, 150, 160, 170, 180, 190, 200, 210, 220, 230, 240,
                 250, 255, 250, 240, 230, 220, 210, 200, 190, 180, 170, 160,
                 150, 140, 130];
    this.atColor = 0;
    this.moveColor = 1;
    this.mess = "Brad & Dan's \"Der verloren Koffer\"";
  }

  async loadData() {
    this.town = await loadImage("assets/intro.gif");
    this.anthem = new Snd("assets/intro.wav");
    this.anthem.loop();
    this.enableNextLevel();
    this.speed = 150;
    this.start();
  }

  cleanUp() {
    super.cleanUp();
    if (this.anthem) this.anthem.stop();
  }

  drawStatic(g) {
    g.drawImage(this.town, 0, 0);
  }

  drawNormal(g) {
    g.font = "italic bold 22px 'Times New Roman', serif";
    const m = g.measureText(this.mess);
    const tw = m.width;
    const tha = m.fontBoundingBoxAscent || 17;
    const drx = Math.floor((400 - tw) / 2);
    const dry = Math.floor((300 + tha / 2) / 2) - 50;

    let ax = drx;
    let tc = this.atColor;
    const half = Math.floor(this.mess.length / 2);

    for (const ch of this.mess) {
      const shift = this.yshift[tc % this.yshift.length];
      g.fillStyle = "#000";
      g.fillText(ch, ax + 2, dry + shift + 2);
      const green = Math.floor(1.7 * (this.cols[(tc + half) % this.cols.length] - 130));
      g.fillStyle = `rgb(234,${green},0)`;
      g.fillText(ch, ax, dry + shift);

      tc++;
      ax += g.measureText(ch).width;
    }

    this.atColor += this.moveColor;
  }
}

File: intro.java
import java.awt.*;
import java.applet.*;

public class intro extends klevel
{
    static final Font f=new Font("TimesRoman",Font.ITALIC|Font.BOLD,22);
    int drx, dry; // base text cords

    int[] yshift = {-6,-6,-5,-4,-3,-2,-1,0,0,-1,-2,-3,-4,-5,-6};
    int[] cols = {130,140,150,160,170,180,190,200,210,220,230,240,250,255,250,240,230,220,210,200,190,180,170,160,150,140,130};
    int atColor=0, moveColor=1;
    int lastCursor=-1;

    FontMetrics fm;

    String mess = "Brad & Dan's \"Der verloren Koffer\"";
    char lets[];

    AudioClip anthem;

    Image town = null;

    public intro ()
    {
        plot = "Willkommen!";
        inst = "Click the NEXT button to go to the next level.";
    }

    public void loadData ()
    {
        town = waitImage("intro.gif");
        anthem = waitAudioClip("intro.au");

        setFont(f);
        lets=mess.toCharArray();

        anthem.loop();
        enableNextLevel();

        loaded=true;
        speed=150;
        start();
    }

    public void cleanUp ()
    {
        anthem.stop();
        anthem = null;
    }

    public void drawNormal (Graphics g)
    {
        fm=g.getFontMetrics();
        int tw  = fm.stringWidth(mess);
        int tha = fm.getAscent();
        drx=(koffer.gw-tw)/2;
        dry=(koffer.gh+tha/2)/2-50;

        int ax=drx;
        int tc=atColor;
        int cv, i;

        for (i=0;i<lets.length;i++) {

            cv=cols[tc % cols.length];
            g.setColor(Color.black);
            g.drawString(String.valueOf(lets[i]),ax+2,dry+yshift[tc % yshift.length]+2);
            g.setColor(new Color(234,(int)(1.7d*((double)cols[(tc+lets.length/2) % cols.length]-130d)),0));
            g.drawString(String.valueOf(lets[i]),ax,dry+yshift[tc % yshift.length]);

            tc++;
            ax+=fm.charWidth(lets[i]);
        }

        atColor+=moveColor;
    }

    public void drawStatic (Graphics g)
    {
        g.drawImage(town,0,0,this);
    }

}